File: solutions/zig/isogram/1/isogram.zig
const std = @import("std");

pub fn isIsogram(str: []const u8) bool {
    var letters: [26]bool = undefined;
    for (str) |c| {
        if (std.ascii.isAlphabetic(c)) {
            const i = std.ascii.toLower(c) - 'a';
            if (letters[i])
                return false;
            letters[i] = true;
        }
    }
    return true;
}

File: solutions/go/isogram/1/isogram.go
// Package isogram should have a package comment that summarizes what it's about.
package isogram

import (
	"unicode"
)

// IsIsogram should have a comment documenting it.
func IsIsogram(word string) bool {
	letters := make(map[rune]bool)
	for _, letter := range word {
		if unicode.IsLetter(letter) {
			if !letters[unicode.ToLower(letter)] {
				letters[unicode.ToLower(letter)] = true
			} else {
				return false
			}
		}
	}
	return true
}
